File: Data Structures & Algorithms/trapping-rain-water/submission-0.java
class Solution {
    public int trap(int[] height) {
        int leftMax = 0;
        int rightMax = 0;
        int left = 0;
        int right = height.length-1;
        int res = 0;
        while(left<right){
            if(height[left]<=height[right]){
                if(leftMax<height[left]){
                    leftMax = height[left];
                }
                res+=leftMax-height[left];
                left++;
            }
            else{
                if(rightMax<height[right]){
                    rightMax = height[right];
                }
                res+=rightMax-height[right];
                right--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        max_left = float('-inf')
        max_right = float('-inf')
        s = 0
        while left<right:
            if height[left]<=height[right]:
                max_left = max(max_left,height[left])
                s+=max_left-height[left]
                left+=1
            else:
                max_right = max(max_right,height[right])
                s+=max_right-height[right]
                right-=1
        return s